secboot/secboot_hooks.go

```diff
@@ -118,21 +122,45 @@ func SealKeysWithFDESetupHook(runHook fde.RunSetupHookFunc, keys []SealKeyReques
 	return nil
 }
 
-func isV1EncryptedKeyFile(p string) bool {
-	// XXX move some of this to kernel/fde
-	var v1KeyPrefix = []byte("USK$")
+func NewKeyDataFromV1HookFile(path string) (*sb.KeyData, error) {
+	sealedKey, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read sealed key file: %v", err)
+	}
 
-	f, err := os.Open(p)
+	handle, err := json.Marshal(sealedKey)
 	if err != nil {
-		return false
+		return nil, err
 	}
-	defer f.Close()
 
-	buf := make([]byte, len(v1KeyPrefix))
-	if _, err := io.ReadFull(f, buf); err != nil {
-		return false
+	params := sb.KeyParams{
+		Handle:       json.RawMessage(handle),
```

Comment:
can you leave a comment stating that we intentionally store key data in the handle as it is treated as a opaque blob by secboot?